### PR TITLE
delete saved root URL when configuration changed

### PR DESCRIPTION
**Pull request by kosgrz**
#### Summary
This PR deletes a saved root URL every time the configuration is changed.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-plugin-skype4business/issues/34

### CODE REVIEW

File: server/configuration.go [hunk 1]
```diff
@@ -89,6 +89,10 @@ func (p *Plugin) OnConfigurationChange() error {
 		return errors.Wrap(err, "failed to load plugin configuration")
 	}
 
+	if err := p.API.KVDelete(RootURLKey); err != nil {
```
Comment (levb) [inline server/configuration.go]: I am not sure exactly how this works, but isn't deleting (resetting/rediscovering) every time in OnConfigurationChange a bit expensive? It is my understanding that the OnConfigurationChange hook may be invoked on an unrelated configuration change.

Comment (kosgrz) [reply server/configuration.go]: Indeed, it should be deleted only when the `domain` setting is changed. Do you know how to fetch the older configuration values in the `OnConfigurationChange` hook?

Comment (mickmister) [reply server/configuration.go]: During the `OnConfigurationChange` hook, you should have access to the current config stored in the plugin process right? You can do a diff between the two configs to find out if the `Domain` value changed.

Comment (kosgrz) [reply server/configuration.go]: Thanks, done :+1:

**Comment by levb** [comment]
@DHaussermann I am not sure if it's easily testable, the change is covered by the unit test.

File: server/configuration.go [hunk 2]
```diff
@@ -89,6 +89,12 @@ func (p *Plugin) OnConfigurationChange() error {
 		return errors.Wrap(err, "failed to load plugin configuration")
 	}
 
+	if configuration.Domain != p.getConfiguration().Domain {
+		if err := p.API.KVDelete(RootURLKey); err != nil {
+			return errors.Wrap(err, "failed to delete saved root URL in KV Store")
+		}
```
Comment (hanzei) [inline server/configuration.go:95]: 0/5 thought: Instead of storing `RootURLKey` in `getRootURL` does it make sense to just update the stored method here?

**Comment by hanzei** [comment]
Skipping QA review per https://community.mattermost.com/core/pl/k1js57wa8tnr5qtrzturas7txh
